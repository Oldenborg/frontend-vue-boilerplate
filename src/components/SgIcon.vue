<template>
  <span :class="prefix">{{type}}</span>
</template>

<script>
export default {
  name: 'SgIcon',
  props: {
    type: {
      type: String,
      required: true,
    },
    prefix: {
      type: String,
      default: 'material-icons',
    },
  },
};
</script>
