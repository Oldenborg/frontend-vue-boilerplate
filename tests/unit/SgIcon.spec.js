import { mount } from '@vue/test-utils';
import SgIcon from '@/components/SgIcon.vue';

describe('SgIcon', () => {
  describe('Properties', () => {
    it('has a required type property', () => {
      const wrapper = mount(SgIcon, {
        propsData: {
          type: 'user',
        },
      });
      expect(wrapper.vm.$options.props.type.type).toBe(String);
      expect(wrapper.vm.$options.props.type.required).toBeTruthy();
    });

    it('has a prefix property with a default value', () => {
      const wrapper = mount(SgIcon, {
        propsData: {
          type: 'user',
        },
      });
      expect(wrapper.vm.$options.props.prefix.type).toBe(String);
      expect(wrapper.vm.$options.props.prefix.default).toBe('material-icons');
    });
  });

  it('renders correctly', () => {
    const wrapper = mount(SgIcon, {
      propsData: {
        type: 'user',
        prefix: 'icon',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
