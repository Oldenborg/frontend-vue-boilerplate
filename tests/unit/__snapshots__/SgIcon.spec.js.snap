// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SgIcon renders correctly 1`] = `<span class="icon">user</span>`;
